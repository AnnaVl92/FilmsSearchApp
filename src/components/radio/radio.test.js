import React from 'react';
import Radio from './radio.jsx';
import { shallow, mount } from "enzyme";

describe('Radio', function() {
	itx('check if radio component render properly',() => {
		const wrapper = shallow(<Radio id="sortReleaseDate" name="sort" value="releaseDate" labelText="release date" />);
		expect(wrapper).toMatchSnapshot();
	});

	itx('test chooseSort with releaseDate value', () => {
		const onChange = jest.fn(),
			component = shallow(<Radio onChange={onChange} />);
		component.find("input").simulate("change","releaseDate");
		expect(onChange).toBeCalledWith("releaseDate");
	});

	itx('test chooseSort with rating value', () => {
		const onChange = jest.fn(),
			component = shallow(<Radio onChange={onChange} />);
		component.find("input").simulate("change","rating");
		expect(onChange).toBeCalledWith("rating");
	});

	itx('test changeSearchFilter with genre value', () => {
		const onChange = jest.fn(),
			component = shallow(<Radio onChange={onChange} />);
		component.find("input").simulate("change","genre");
		expect(onChange).toBeCalledWith("genre");
	});

	itx('test changeSearchFilter with title value', () => {
		const onChange = jest.fn(),
			component = shallow(<Radio onChange={onChange} />);
		component.find("input").simulate("change","title");
		expect(onChange).toBeCalledWith("title");
	});
});