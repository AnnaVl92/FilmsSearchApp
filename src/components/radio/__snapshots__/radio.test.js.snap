// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio check if radio component render properly 1`] = `
<div
  className="form-check"
>
  <input
    className="form-check-input"
    id="sortReleaseDate"
    name="sort"
    type="radio"
    value="releaseDate"
  />
  <label
    className="form-check-label"
    htmlFor="sortReleaseDate"
  >
    release date
  </label>
</div>
`;
